### PR TITLE
test(melange): remove rescript-syntax tests from CI

### PR DESCRIPTION
**Pull request by rgrinberg**
Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

**Comment by rgrinberg** [comment]
@gridbugs does the monorepo benchmark need to be updated to 3.12?

**Comment by gridbugs** [comment]
> @gridbugs does the monorepo benchmark need to be updated to 3.12?

I don't think so. Looks like it ran successfully.

**Comment by gridbugs** [comment]
> @gridbugs does the monorepo benchmark need to be updated to 3.12?

The error is:
```
Step 24/28 : RUN . ~/.profile && dune build bin/main.exe --release
 ---> Running in 1113cef47d40
File "dune-project", line 1, characters 11-15:
1 | (lang dune 3.12)
               ^^^^
Error: Version 3.12 of the dune language is not supported.
Supported versions of this extension in version 3.12 of the dune language:
- 1.0 to 1.12
- 2.0 to 2.9
- 3.0 to 3.11
```

That error message doesn't make sense to me. Which extension is it talking about? Is the dune language itself an extension of dune?

**Comment by rgrinberg** [comment]
>  Is the dune language itself an extension of dune?

That's what it claims but it doesn't really work like that. We should amend that. Anyway, the version of dune used to build the benchmarks is too old to understand `(lang dune 3.12)` after this PR.

**Comment by gridbugs** [comment]
Ah so the failing step is when it builds dune in the benchmark docker container. I think it's using a version of dune that's too old. The installed version of dune in the cached docker image is too old.

**Comment by gridbugs** [comment]
https://github.com/ocaml/dune/pull/9891 will hopefully fix the problem.

**Review by anmonteiro — APPROVED** [review]

File: test/blackbox-tests/test-cases/melange/dune [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 (cram
  (deps %{bin:rescript_syntax})
- (applies_to rescript-syntax))
+ (applies_to rescript-syntax)
+ (runtest_alias false))
```
Comment (jchavarri) [inline test/blackbox-tests/test-cases/melange/dune:8]: What's the upside of this vs removing the `cram` stanza altogether?

**Review by jchavarri — APPROVED** [review]
lgtm. Is this part still needed?

https://github.com/ocaml/dune/blob/e06a128a52b47b884e85c0470ba4173ed64b9501/src/dune_engine/action.ml#L77-L78

**Comment by rgrinberg** [comment]
> lgtm. Is this part still needed?
> 
> https://github.com/ocaml/dune/blob/e06a128a52b47b884e85c0470ba4173ed64b9501/src/dune_engine/action.ml#L77-L78

Indeed we can remove it. Although I'd rather think of a way of removing all the hints from the engine.

File: Makefile [hunk 2]
```diff
@@ -25,8 +25,7 @@ ppx_inline_test \
 ppxlib \
 ctypes \
 "utop>=2.6.0" \
-"melange>=1.0.0" \
-"rescript-syntax"
+"melange=1.0.0"
```
Comment (emillon) [inline Makefile:28]: should have been >=1.0.0, right?

Comment (emillon) [reply Makefile:28]: ah no, the tests are using 1.0.0 only features.

**Pull request merged**